jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -1690,17 +1690,13 @@ public byte[] takeByteArray()
             checkNotReleased();
             return switch (_buffers.size())
             {
-                case 0 -> BufferUtil.EMPTY_BUFFER.array();
+                case 0 -> BufferUtil.EMPTY_BYTES;
                 case 1 ->
                 {
                     RetainableByteBuffer buffer = _buffers.get(0);
                     _aggregate = null;
                     _buffers.clear();
-
-                    // The array within the buffer can be used if it is not pooled, is not shared and it exits
-                    byte[] array = (!(buffer instanceof Pooled) && !buffer.isRetained() && !buffer.isDirect())
-                        ? buffer.getByteBuffer().array() : BufferUtil.toArray(buffer.getByteBuffer());
-
+                    byte[] array = BufferUtil.toArray(buffer.getByteBuffer());
```

Comment:
What was the problem with taking a non pooled array without copying?

If it really doesn't work, then the javadoc above regarding the length needs to be updated

Reply:
With regards to length, the code was just following the javadoc above, which allows for a longer array than the content to be returned.   So if you want to make this method always copy, then you SHOULD update the javadoc.

For the offset, yes it should check that the offset is zero (as it mostly is).

So either update the javadoc OR add the zero offset check.  Don't do neither.